### PR TITLE
CUMULUS-1682 Reduce code complexity with ESLint

### PR DESCRIPTION
**Pull request by nemreid**
**Summary:** Summary of changes

Addresses [CUMULUS-1682: Use ESLint complexity functionality to find highly complex code to simplify and simplify it.](https://bugs.earthdata.nasa.gov/browse/CUMULUS-1682)

## Changes

* Enforce complexity 10 with ESLint

## PR Checklist

- [ ] Update CHANGELOG
- [ ] Unit tests
- [ ] Adhoc testing
- [ ] Integration tests

### CODE REVIEW

File: packages/ingest/granule.js [hunk 1]
```diff
@@ -587,6 +587,38 @@ class HttpGranule extends httpMixin(baseProtocol(Granule)) {}
  */
 class S3Granule extends s3Mixin(baseProtocol(Granule)) {}
 
+function selectGranuleDiscoverer(protocol) {
```
Comment (markdboyd) [inline packages/ingest/granule.js]: About to be irrrelevant but I guess that doesn't matter

File: packages/ingest/parse-pdr.js [hunk 2]
```diff
@@ -40,6 +40,40 @@ function getItem(spec, pdrName, name, must = true) {
...
 
+/**
+ * Validate parsedSpec matches expectations
+ *
+ * @param {Object} spec - parsed spec
+ * @returns {Object} validated spec
+ * @throws {Error}
+ */
+function validateParsedSpec(spec) {
+  const parsedSpec = Object.assign({}, spec);
```
Comment (markdboyd) [inline packages/ingest/parse-pdr.js]: Why are we passing in the whole spec if we're just validating the checksum properties?

File: packages/ingest/parse-pdr.js [hunk 3]
```diff
@@ -40,6 +40,40 @@ function getItem(spec, pdrName, name, must = true) {
...
 }
 
+/**
+ * Validate parsedSpec matches expectations
+ *
+ * @param {Object} spec - parsed spec
+ * @returns {Object} validated spec
+ * @throws {Error}
+ */
+function validateParsedSpec(spec) {
```
Comment (markdboyd) [inline packages/ingest/parse-pdr.js]: Would `validateChecksumInformation` be a better name for what this function is doing?

**Comment by nemreid** [comment]
I'm going to separate out the recursion fix and re-do this PR later since it will definitely be affected by Marc's big ingest refactor.

**Pull request closed without merging**